### PR TITLE
ProxyTxn refactor: moved host_res_style to t_state.dns_info

### PR DESCRIPTION
**Pull request by a-canary**
Organizing data before refactoring. Moved ProxyTxn::host_res_style to t_state.dns_info, because that makes sense to me.

### CODE REVIEW

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -494,6 +494,12 @@ HttpSM::attach_client_session(ProxyTransaction *client_vc, IOBufferReader *buffe
     debug_on = true;
   }
 
+  ink_release_assert(ua_txn->get_proxy_ssn());
```
Comment (SolidWallOfCode) [inline proxy/http/HttpSM.cc]: I think these should be debug asserts.

Comment (a-canary) [reply proxy/http/HttpSM.cc]: I can change it to that. Do debug asserts execute in Autests and regression tests?

Comment (SolidWallOfCode) [reply proxy/http/HttpSM.cc]: Yes.

**Comment by zwoop** [comment]
@a-canary This has merge conflicts.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**